### PR TITLE
Special Metadata Types section is added

### PR DESCRIPTION
**Pull request by ArqumFarooqui-NEC**
Included the section for Special Metadata Types for issue #3434

### CODE REVIEW

**Comment by ArqumFarooqui-NEC** [comment]
Hi @fgalan sir, 
I've fixed the comments for issue #3434 provided on PR #3652

File: doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib [hunk 1]
```diff
@@ -354,6 +355,34 @@ The list of builtin attributes is as follows:
 Like regular attributes, they can be  used in `q` filters and in `orderBy`.
 However, they cannot be used in resource URLs.
 
+## Special Metadata Types
+
+Generally speaking, user-defined metadata types are informative; they are processed by the NGSIv2
+server in an opaque way. Nonetheless, the types described below are used to convey a special
+meaning:
+
+* `DateTime`:  identifies dates, in ISO8601 format. This Metadata can be used with the query
```
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib:364]:
```suggestion
* `DateTime`:  identifies dates, in ISO8601 format. This metadata can be used with the query
```

File: doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib [hunk 2]
```diff
@@ -354,6 +355,34 @@ The list of builtin attributes is as follows:
...
+```
+
+The list of Special Metadata Types is as follows:
+
+* `dateCreated` (type: `DateTime`): attribute creation date as an ISO 8601 string.
+
+* `dateModified` (type: `DateTime`): attribute modification date as an ISO 8601 string.
+
+* `dateExpires` (type: `DateTime`): attribute expiration date as an ISO 8601 string. How the server
+  controls attribute expiration is an implementation specific aspect.
```
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib:384]: Looking to "Special Attribute Types" section (around L313-L314) I suggest to remove this block of line.

Comment (fgalan) [reply doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib:384]: I mean the block from L377-L384

File: doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib [hunk 3]
```diff
@@ -39,6 +39,7 @@ This specification is the final and stable version of NGSIv2 API specification (
 
 # Changelog
 
+Add "Special Metadata Types" section
 Changes since RC-2018.07:
```
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib:43]:
```suggestion
Add "Special Metadata Types" section

Changes since RC-2018.07:
```

**Comment by fgalan** [comment]
I have realized that you have done the changes in fiware-ngsiv2-2.0-2018_09_15.apib file. That file shouldn't be changed, as it corresponds to the frozen NGSI 2.0.0 version and it should not be touched.

Please apply the changes to fiware-ngsiv2-2.0.1-wip.apib file and leave fiware-ngsiv2-2.0-2018_09_15.apib untouched.

Thanks!

File: doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib (hunk 1 above)
Comment (ArqumFarooqui-NEC) [reply doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib]: Fixed in 4f24cda

File: doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib (hunk 2 above)
Comment (ArqumFarooqui-NEC) [reply doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib]: Fixed in 4f24cda

File: doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib (hunk 3 above)
Comment (ArqumFarooqui-NEC) [reply doc/apiary/v2/fiware-ngsiv2-2.0-2018_09_15.apib]: Fixed in 4f24cda

**Comment by ArqumFarooqui-NEC** [comment]
> I have realized that you have done the changes in fiware-ngsiv2-2.0-2018_09_15.apib file. That file shouldn't be changed, as it corresponds to the frozen NGSI 2.0.0 version and it should not be touched.
> 
> Please apply the changes to fiware-ngsiv2-2.0.1-wip.apib file and leave fiware-ngsiv2-2.0-2018_09_15.apib untouched.
> 
> Thanks!

Hi @fgalan sir,
I've updated the `fiware-ngsiv2-2.0.1-wip.apib` file adding new section for `Special Metadata Types` and undo all the changes in `fiware-ngsiv2-2.0-2018_09_15.apib`.

**Comment by Chandradeep-NEC** [comment]
@fgalan 
Please provide your feedback on updated patch. If it looks ok then please merge it to master.

**Comment by Chandradeep-NEC** [comment]
Dear @fgalan 

Is there any update on this PR?

**Comment by fgalan** [comment]
I did a small test to check that Orion is actually honoring the change done in the documentation in this PR. In particular, with this entity creation:

```
curl localhost:1026/v2/entities -s -S -H 'Content-Type: application/json' -d @- <<EOF
{
  "id": "E1",
  "type": "T",
  "temperature": {
    "value": "2020-06-10T10:42:11Z",
    "type": "DateTime",
	"metadata": {
	  "M1": {
	    "value": "2020-06-10T10:42:11Z",
		"type": "DateTime"
	  }
	}
  }
}
EOF
```

The document created in DB is this one:

```
{
	"_id" : {
		"id" : "E1",
		"type" : "T",
		"servicePath" : "/"
	},
	"attrNames" : [
		"temperature"
	],
	"attrs" : {
		"temperature" : {
			"type" : "DateTime",
			"creDate" : 1592295125,
			"modDate" : 1592295125,
			"value" : 1591785731,
			"md" : {
				"M1" : {
					"type" : "DateTime",
					"value" : 1591785731
				}
			},
			"mdNames" : [
				"M1"
			]
		}
	},
	"creDate" : 1592295125,
	"modDate" : 1592295125,
	"lastCorrelator" : "1093238a-afa9-11ea-b166-000c29df7908"
}
```
and:

imagen



So it seems fine and this PR can be merged. Thanks!

**Comment by fgalan** [comment]
@fisuda I don't remember if there is an equivalent of doc/apiary/v2/fiware-ngsiv2-2.0.1-wip.apib in the Japanese translation. In positive case, it would be great if you could update it based on the content of this PR. Thanks!

**Comment by fgalan** [comment]
@fisuda please take also this last-minute-typo-fix into account: PR https://github.com/telefonicaid/fiware-orion/pull/3664

**Comment by fisuda** [comment]
@fgalan , Noted with thanks. I will check if it is necessary to update fiware-ngsiv2-2.0.1-wipe.apib in the Japanese translation.

**Pull request merged**